File: src/commonMain/kotlin/com/scoreo/domain/port/GameTypeRepository.kt
package com.scoreo.domain.port

import com.scoreo.domain.model.GameType

interface GameTypeRepository {
    fun getAll(): List<GameType>
    fun save(gameType: GameType)
    fun saveAll(gameTypes: List<GameType>)
    fun findById(id: String): GameType?
}

File: src/domain/port/gameTypeRepository.ts
import type { GameType } from '../model/gameType'

export interface GameTypeRepository {
  getAll(includeInactive?: boolean): GameType[]
  save(gameType: GameType): void
  saveAll(gameTypes: GameType[]): void
  findById(id: string): GameType | undefined
  /** Permanent removal, unlike the `active = false` soft-delete used by archiving. */
  hardDelete(id: string): void
  deleteAll(): void
}
